src/check_executor.rs

```diff
@@ -103,55 +115,78 @@ impl CheckResult {
 
 pub fn run_executor(
     concurrency: usize,
+    failure_retries: u16,
     checker: Arc<impl Checker + 'static>,
     producer: Arc<impl ResultsProducer + 'static>,
     region: String,
-) -> (CheckSender, JoinHandle<()>) {
+) -> (Arc<CheckSender>, JoinHandle<()>) {
     tracing::info!("executor.starting");
 
-    let (check_sender, receiver) = CheckSender::new();
-    let queue_size = check_sender.queue_size.clone();
-    let num_running = check_sender.num_running.clone();
+    let (sender, reciever) = CheckSender::new();
+    let queue_size = sender.queue_size.clone();
+    let num_running = sender.num_running.clone();
+
+    let check_sender = Arc::new(sender);
+    let executor_check_sender = check_sender.clone();
 
-    let executor = tokio::spawn(async move {
+    let conf = ExecutorConfig {
+        concurrency,
+        failure_retries,
+        region,
+    };
```

Comment:
small cleanup, clippy was complaining that we're giving the `executor_loop` function too many arguments, so just moved some into a config type thing